ivy/utils/_importlib.py

```diff
@@ -1,11 +1,95 @@
+import os
 import sys
+from typing import List, Set
+from pathlib import Path
 from importlib.util import resolve_name, module_from_spec
 from ivy.utils.backend import ast_helpers
 
 
 import_cache = {}
 path_hooks = []
 
+ivy_path_abs = Path(sys.modules["ivy"].__file__).parents[1]
+
+# Note that any modules listed as 'to skip' should not depend on the Ivy backend state.
+# If they do, the behavior of ivy.with_backend is undefined and may not function as
+# Expected.
+MODULES_TO_SKIP = ["ivy.compiler"]
+MODULES_TO_EXCLUDE = []
+
+
+def _get_modules(absolute_name: str) -> Set[str]:
+    """Get all Ivy modules and sub-packages in a module
+
+    Parameters
+    ----------
+    absolute_name
+        Full absolute name of the module
+
+    Returns
+    -------
+        A set of modules and sub-packages found
+    """
+    if not absolute_name.startswith("ivy."):
+        raise ValueError("Module name must be specified using full namespace.")
+
+    name_to_path = absolute_name.replace(".", os.path.sep)
+    module_path = os.path.join(ivy_path_abs, name_to_path)
+
+    modules = set()
+    modules.add(absolute_name)
+
+    for root, dirs, files in os.walk(module_path):
+        if root.endswith("__"):
+            continue
+        common = os.path.commonpath([root, module_path])
+        full_name = absolute_name + root[len(common) :].replace(os.path.sep, ".")
+        for mod_name in files:
+            if mod_name.startswith("__") or not mod_name.endswith(".py"):
+                continue
+            modules.add(f"{full_name}.{mod_name[:-3]}")
```

Comment:
You can turn this into one line
```python
modules.update([f"{full_name}.{mod_name[:-3]}" for mod_name in files
                         if not mod_name.startswith("__") and mod_name.endswith(".py")])
                 
```
Whether this is more readable or not is up to debate tbh, also a comment that explains the `[:-3]` could be slightly helpful (it is apparent from the context that it excludes the .py but it is always nice to explicitly mention it)

Reply:
I agree with the comment part, for the second part I think we can keep it as a for loop for this once as it may be a little overkill (too much characters), will also stick with for loop for the second one to be more consistent!